src/Orleans/Utils/WrappedException.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Orleans.Runtime.Internal
+{
+    /// <summary>Exception used to communicate with the storage provider, so that it throws this exception to its caller.</summary>
+    [Serializable]
+    public class WrappedException : OrleansException
```

Comment:
BTW, the memory storage grain itself is also an implementation detail of this provider that is public, so that was my encouragement to go with this option, but I certainly prefer making this new exception internal